core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -157,12 +159,23 @@ private void validateVolumeType(VolumeTemplate value, Platform platform, Validat
         }
     }
 
+    private boolean isIDBrokerInstanceGroup(InstanceGroup instanceGroup) {
+        return GROUP_NAME_ID_BROKER.equalsIgnoreCase(instanceGroup.getGroupName());
+    }
+
     private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeCount() > config.maximumNumber()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeCount() > 0) {
```

Comment:
why we are doing this? we dont want to set it if the user specify us. QA team can test any scenario.

Reply:
@doktoric I think we never want to attach a data volume for the IDBroker node. I don't think QA team should or want to test this. This is applicable ONLY for IDBroker nodes. The reason is that IDBroker should not need a data volume.